packages/@react-spectrum/tabs/src/Tabs.tsx

```diff
@@ -105,6 +105,12 @@ export function Tabs<T extends object>(props: SpectrumTabsProps<T>) {
 
   useResizeObserver({ref: wrapperRef, onResize: checkShouldCollapse});
 
+  // When the tabs are collapsed, the tabPanel should be labelled by the Picker button element.
+  let collapsibleTabListId = useId();
+  if (collapse) {
```

Comment:
Tested in the orientation flip story and the collapse id aria-labelledby remained when the orientation flipped. Its a bit of a non-sensical example but you could fix it by adding a check here on `orientation !== 'vertical'` as well